src/nvim/memline.c

```diff
@@ -1522,11 +1521,8 @@ static time_t swapfile_info(char_u *fname)
 #endif
     MSG_PUTS(_("             dated: "));
     x = file_info.stat.st_mtim.tv_sec;
-    p = ctime(&x);  // includes '\n'
-    if (p == NULL)
-      MSG_PUTS("(invalid)\n");
-    else
-      MSG_PUTS(p);
+    char ctime_buf[50];
+    MSG_PUTS(os_ctime_r(&x, ctime_buf));
```

Comment:
I diverge from Vim here by using `os_ctime_r` directly because `get_ctime` uses a static buffer. I assume that a static buffer makes a function non-thread-safe unless a thread explicitly creates it in runtime or each thread has a separate static buffer (indexed by thread id?).